arwen/host/execution_test.go

```diff
@@ -262,6 +264,91 @@ func TestExecution_ManyDeployments(t *testing.T) {
 	}
 }
 
+func TestExecution_MultipleArwens_OverlapingContractInstanceData(t *testing.T) {
+	code := GetTestSCCode("counter", "../../")
+
+	input := DefaultTestContractCallInput()
+	input.GasProvided = 1000000
+	input.Function = get
+
+	host1, instanceRecorder1 := defaultTestArwenForCallWithInstanceRecorderMock(t, code, nil)
+	runtimeContextMock := contextmock.NewRuntimeContextWrapper(&host1.runtimeContext)
+	runtimeContextMock.CleanWasmerInstanceFunc = func() {}
+	host1.runtimeContext = runtimeContextMock
+
+	vmOutput, err := host1.RunSmartContractCall(input)
+	require.Nil(t, err)
+	require.NotNil(t, vmOutput)
+
+	var host1InstancesData = make(map[interface{}]bool)
+	for _, instance := range instanceRecorder1.GetContractInstances(code) {
+		host1InstancesData[instance.GetData()] = true
```

Comment:
Only one instance is created by `host1`, so this `for` loop can be replaced by:
```
host1InstancesData := instance.GetData()
```
(see comment below on how to extract the `instance` from a host.